metrics.go

```diff
@@ -0,0 +1,266 @@
+package frankenphp
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"path/filepath"
+	"regexp"
+	"sync"
+	"time"
+)
+
+type Metrics interface {
```

Comment:
Couldn't we make everything metrics-related unexported instead? This extension point looks useless because:

- adding more FrankenPHP metrics will require changing the internal code anyway 
- adding other metrics is already possible just by registering an extra custom exporter (and a Caddy module if using Caddy).

I would remove this interface and just provide a `WithMetricsEnabled(bool)` option to reduce the API surface.

WDYT?

Reply:
Hmmm. I had to think about this one for a bit.

> adding more FrankenPHP metrics will require changing the internal code anyway

This extension point isn't really for adding more metrics, but for passing in something that knows how to count. For example, if you wanted to use Frankenphp as a library and create a StatsDMetrics provider to send metrics to statsd instead of prometheus.

Then you need to define it and send it with `WithMetrics(myStatsDCollector)` or whatever.

> adding other metrics is already possible just by registering an extra custom exporter (and a Caddy module if using Caddy).

Totally agree, but then it makes Frankenphp dependent upon how caddy collects metrics or locks it into prometheus. I don't think there is anything wrong with that, if that is the direction you want to go. But I left it open in this implementation.

Reply:
To be honest I would prefer starting with the smallest possible API, and if there is a need, we could add new methods exposing some counters to third-party libraires. WDYT?

Reply:
Hmm, in Mercure we followed the same
strategy as yours. https://github.com/dunglas/mercure/blob/main/metrics.go

Let's keep this as-is for consistency between the projects then.

Could you just rename NullMetrics in NopMetrics (Zap uses "nop" too)?